cmd/uhc/cluster/login/login.go

```diff
@@ -152,3 +168,69 @@ func run(cmd *cobra.Command, argv []string) {
 		os.Exit(1)
 	}
 }
+
+// doSurvey will ask user to choose one if there are more than one clusters match the query
+func doSurvey(clusters []*v1.Cluster) (cluster *v1.Cluster, err error) {
+	clusterList := []string{}
+	for _, v := range clusters {
+		clusterList = append(clusterList, fmt.Sprintf("Name: %s, ID: %s", v.Name(), v.ID()))
+	}
+	choice := ""
+	prompt := &survey.Select{
+		Message: "Please choose a cluster:",
+		Options: clusterList,
+		Default: clusterList[0],
+	}
+	err = survey.AskOne(prompt, &choice, func(ans interface{}) error {
+		choice := ans.(string)
+		found := false
+		for _, v := range clusters {
+			if strings.Contains(choice, v.ID()) {
+				found = true
+				cluster = v
+			}
+		}
+		if !found {
+			return fmt.Errorf("the cluster you choose is not valid: %s", choice)
+		}
+		return nil
+	})
+	return cluster, err
+}
+
+// findClusters finds the clusters that match the given key. A cluster matches the key if its
+// identifier is that key, or if its name starts with that key. For example, the key `prd-2305`
+// doesn't match a cluster directly because it isn't a valid identifier, but it matches all clusters
+// whose names start with `prd-2305`.
+func findClusters(collection *v1.ClustersClient, key string) (clusters []*v1.Cluster, err error) {
+
+	// Get the resource that manages the cluster that we want to display:
+	clusterResource := collection.Cluster(key)
+	response, err := clusterResource.Get().Send()
+
+	// If it's not an cluster id, try to query clusters using search param, we assume the
+	// list should be within 50, if not, we need update the cluster names prefix to minimize
+	// the list
+	if err != nil {
+		pageSize := 50
+		pageIndex := 1
+		listRequest := collection.List().
+			Size(pageSize).
+			Page(pageIndex)
+		listRequest.Search("name like '" + key + "'")
+		listResponse, err := listRequest.Send()
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Can't retrieve clusters: %s\n", err)
+			return clusters, err
+		}
+		listResponse.Items().Each(func(cluster *v1.Cluster) bool {
+			clusters = append(clusters, cluster)
+			return true
+		})
```

Comment:
Note that here you are limiting the result to the first 50 matching clusters. There may be more than that, and you will be silently ignoring them. Is that what you want?

Reply:
should I exit(1), and print "thre are more than 50 clusters match the condition, please use more accurate search param" , WDYT?

Reply:
Yes, that seems reasonable. To do it you can use the `total` value returned by the request. That is the total number of items that satisfy the condition, regardless of the page size. So you can do something like this:

```go
size := listResponse.Size()
total := listResponse.Total()
if total > size {
        ...
}
```

It may be a good ideal to add that to as an output parameter of the `findClusters` function:

```go
func findClusters(...) (clusters []*v1.Cluster, total int, err error) {
        ...
}
```

That way you can still keep the user interface logic out of the function, and use the result to print that message that you suggest. Something like this:

```go
clusters, total, err := findClusters(...)
if err != nil {
        ...
}
if total > len(clusters) {
        fmt.Fprintf(
                os.Stdout,
                "There are %d clusters that match key '%s', but only the first %d will "+
                        be shown; consider using a more specific key.",
                total, key, len(clusters),
        )
}
```

After printing that message you can exit as you suggest, or you can let the user continue. Both alternatives seem reasonable to me.